### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by MiraAr**

### CODE REVIEW

**Comment by MiraAr** [comment]
PTAL

**Review by emrerdem1 — APPROVED** [review]

File: src/components/auth/login/GoogleAndFacebookLogIn.jsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const facebookProvider = new firebase.auth.FacebookAuthProvider();
+
+  const postData = async (user) => {
+    const info = {
+      name: user.displayName,
+      userId: user.uid,
+      imageUrl: user.photoURL,
+      isAdmin: false,
+    };
+    // console.log(info.userId)
```
Comment (louisrli) [inline src/components/auth/login/GoogleAndFacebookLogIn.jsx]: console

File: src/components/auth/login/GoogleAndFacebookLogIn.jsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    const createNewSubCollection = await db
+      .collection("users")
+      .doc(info.userId)
+      .collection("liked")
+      .add({
+        liked: "0",
```
Comment (louisrli) [inline src/components/auth/login/GoogleAndFacebookLogIn.jsx]: i feel like this should be a number not a string?

File: src/components/auth/login/GoogleAndFacebookLogIn.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const logIn = (provider) => {
+    firebase
+      .auth()
+      .signInWithPopup(provider)
+      .then((result) => {
+        const user = result.user;
+        postData(user);
+      })
+      .catch(function (error) {
+        const errorCode = error.code;
```
Comment (louisrli) [inline src/components/auth/login/GoogleAndFacebookLogIn.jsx]: these variables are all unused

File: src/components/auth/login/GoogleAndFacebookLogIn.jsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        const email = error.email;
+        const credential = error.credential;
+      });
+  };
+
+  return (
+    <div>
+      <button onClick={() => logIn(googleProvider)}> Log in with Google</button>
+      <button onClick={() => logIn(facebookProvider)}>
+        {" "}
```
Comment (louisrli) [inline src/components/auth/login/GoogleAndFacebookLogIn.jsx]: delete this space

File: src/components/auth/login/GoogleAndFacebookLogIn.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        const email = error.email;
+        const credential = error.credential;
+      });
+  };
+
+  return (
+    <div>
+      <button onClick={() => logIn(googleProvider)}> Log in with Google</button>
```
Comment (louisrli) [inline src/components/auth/login/GoogleAndFacebookLogIn.jsx]: extra space before Log

File: src/components/auth/login/Login.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+      .get();
+    console.log(loggedInUserName.data());
+    setIsAdmin(loggedInUserName.data().isAdmin);
+  };
+
+  return (
+    <>
+      {isAdmin && <AddProducts />}
+      <h5>sign in</h5>
+      <form onSubmit={(e) => userLogin(e)}>
```
Comment (louisrli) [inline src/components/auth/login/Login.jsx]: `onSubmit={userLogin}`

File: src/components/auth/login/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+  const userLogout = () => {
+    auth.signOut().then(() => {
+      console.log("user logged out");
+      setIsAdmin(false);
+    });
+  };
+  const userLogin = async (e) => {
+    e.preventDefault();
+    const email = e.target[0].value;
```
Comment (louisrli) [inline src/components/auth/login/Login.jsx:17]: does this work? i didn't know that e.target could be used like this. also console.log in this fn needs to be cleaned up

File: src/components/auth/signUp/SignUp.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      .doc(newUser.user.uid)
+      .set({
+        name: userName,
+        isAdmin: false,
+      });
+  };
+  return (
+    <div>
+      <h5>Sign Up</h5>
+      <form onSubmit={(e) => createNewUser(e)}>
```
Comment (louisrli) [inline src/components/auth/signUp/SignUp.jsx]: same as previous comment

File: src/components/auth/signUp/SignUp.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        <input type="text" id="nameField" placeholder="Name" />
```
Comment (louisrli) [inline src/components/auth/signUp/SignUp.jsx]: remove these ids, generally we never use ids and they're not used anywhere anyway

Comment (louisrli) [reply src/components/auth/signUp/SignUp.jsx]: same with all other ids, use classes instead if you need to

File: src/containers/App.js [hunk 10]
```diff
@@ -1,10 +1,12 @@
 import React from "react";
 import "../App.scss";
+import Home from "../components/home/Home";
 
 function App() {
   return (
     <div className="App">
       <h1>Hello</h1>
```
Comment (louisrli) [inline src/containers/App.js]: delete this line hello

File: src/components/auth/login/Login.jsx (hunk 7 above)
Comment (MiraAr) [reply src/components/auth/login/Login.jsx:17]: we did it this way instead of using the input id.

**Comment by MiraAr** [comment]
PTAL

**Comment by louisrli** [comment]
merge conflicts

**Comment by MiraAr** [comment]
PTAL

**Review by louisrli — APPROVED** [review]

**Pull request merged**